### ISSUE TITLE
Waymo -> Kitti, evaluation error in source

### ISSUE THREAD

**Issue by AbhishekKaushikCV** [CVMI-Lab/ST3D#67]
**Task: Waymo  -> Kitti** [config_file_using](https://github.com/CVMI-Lab/ST3D/blob/master/tools/cfgs/da-waymo-kitti_models/pvrcnn/pvrcnn_old_anchor.yaml)
- I want to get the **results of Waymo validation data in the form of Kitti metric.**
- In the training of PVRCNN with the given Config, I am getting an error in the evaluation of the source domain.

**Error Message:**
`slurmstepd: error: Detected 1 oom-kill event(s) in step 89606.0 cgroup. Some of your processes may have been killed by the cgroup out-of-memory handler.
srun: error: linse18: task 0: Out Of Memory`
 - the training is completed without any error, with the same hardware specifications (two gpus 12GB each)
 - **but when I change the `EVAL_METRIC`  from `kitti` to `waymo` in the config, then there is no error (but I am getting the Waymo Metric)** 
 - I have tried with (two GPUs 24GB each) but still getting same error
 - **Is there a way to get the results of a model in the form of KITTI metric?**
 - did anyone face this error before? if so, how can I solve this error?

**Comment by jihanyang** [comment]
Can you share the running command and log here?

**Comment by AbhishekKaushikCV** [comment]
Here is the command that I used: 

`python train.py --cfg_file cfgs/da-waymo-kitti_models/secondiou/secondiou_car_gt.yaml --batch_size 2 --max_ckpt_save_num 5 --num_epochs_to_eval 5 --eval_src --extra_tag train_secondiou_gt_car_waymo`
- Here is the log file:
[train_secondiou_gt_car_waymo_kittimetric_89765.txt](https://github.com/CVMI-Lab/ST3D/files/10497545/train_secondiou_gt_car_waymo_kittimetric_89765.txt)

**Comment by jihanyang** [comment]
It seems that this log uses waymo metric, and I cannot see the error with kitti metric.

**Comment by AbhishekKaushikCV** [comment]
- i am sorry for that, by mistake I uploaded the different file
- I updated the previous log file with the `kitti` metric log

**Comment by jihanyang** [comment]
It seems that you already finish inference part, and the error shows that you don't have enough memory. I have two suggestion but not sure whether can help.
1. Try to use test.py to evaluate like `python test.py --cfg_file xxx --extra_tag xxx --batch_size xx --ckpt xxx`. Then you don't need to load training set, which will occupy your memory.
2. you can try to directly do the evaluation without inference stage as follow: you just need to load the result.pkl that has been save as `eval_det_annos`. In this manner, you don't need even launch pcdet and load any data into memory
https://github.com/CVMI-Lab/ST3D/blob/cf5ece6e43fc7c5801532847d8e96aded6adf202/pcdet/datasets/kitti/kitti_dataset.py#L334-L344
